protinfo_linux.go

```diff
@@ -19,38 +20,38 @@ func LinkGetProtinfo(link Link) (Protinfo, error) {
 		return pi, err
 	}
 
-	for _, m := range msgs {
-		ans := nl.DeserializeIfInfomsg(m)
+	for i := range msgs {
```

Comment:
I looked this up today and according to the internet it isn't a reference
http://stackoverflow.com/questions/20185511/golang-range-references-instead-values

On Fri, Jul 3, 2015, 1:29 PM Alexander Morozov <notifications@github.com>
wrote:

> In protinfo_linux.go
> <https://github.com/vishvananda/netlink/pull/40#discussion_r33880037>:
>
> > @@ -19,38 +20,38 @@ func LinkGetProtinfo(link Link) (Protinfo, error) {
> >  		return pi, err
> >  	}
> >
> > -	for _, m := range msgs {
> > -		ans := nl.DeserializeIfInfomsg(m)
> > +	for i := range msgs {
>
> msgs has reference type anyway, there is no copy here
> and you should avoid premature optimizations, sacrificing readability
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vishvananda/netlink/pull/40/files#r33880037>.
>